app/src/features/apiClient/screens/apiClient/APIClient.tsx

```diff
@@ -63,10 +65,17 @@ export const APIClient: React.FC<Props> = () => {
       });
   }, [requestId]);
 
-  const addToHistory = useCallback((apiEntry: RQAPI.Entry) => {
-    setHistory((history) => [...history, apiEntry]);
-    addToHistoryInStore(apiEntry);
-  }, []);
+  const addToHistory = useCallback(
+    (apiEntry: RQAPI.Entry) => {
+      const renderedEntry = {
+        ...apiEntry,
+        request: renderVariables<RQAPI.Entry["request"]>(apiEntry.request),
+      };
+      setHistory((history) => [...history, renderedEntry]);
+      addToHistoryInStore(renderedEntry);
+    },
+    [renderVariables]
+  );
```

Comment:
Add to history shouldn't render template again. It should get the rendered API Request directly